iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -72,29 +115,55 @@ static LogicalResult setContractConfig(FuncOp entryPoint, linalg::LinalgOp op) {
   int64_t tileX = 2;
   int64_t tileY = 256;
   int64_t tileK = 4;
+  IREE::Codegen::DispatchLoweringPassPipeline pipeline =
+      IREE::Codegen::DispatchLoweringPassPipeline::LLVMGPUMatmulSimt;
   SmallVector<int64_t, 3> workgroupSize = {2 * cudaWarpSize, 1, 1};
   bool isStaticSize =
       sizeM != ShapedType::kDynamicSize && sizeN != ShapedType::kDynamicSize;
   if (isStaticSize) {
-    // Special case for very small matrices.
-    if (sizeM * sizeN <= cudaWarpSize) {
-      tileX = sizeN;
-      tileY = sizeM;
-      workgroupSize = {sizeM, sizeN, 1};
+    bool foundTensorCoreConfig = false;
+    if (supportsTensorCore(entryPoint, op)) {
```

Comment:
Just in terms of readability, move this to another function?

Reply:
To add, one way to make things more readable is to think of each piece following the logic

```
if (is_lowering_config_set) {
   ...
   set_lowering_config
}
```

Then its a matter of ordering these different pieces. That makes the code more readable IMO.

Reply:
Changed the logic to a sequence of:
```
if(!isLoweringConfigSet) {
 if(match) {
    ...
    isLoweringConfigSet = true;
  }
} 
if(!isLoweringConfigSet) {
```

Reply:
I dont see a point of `isLoweringConfigSet` here either. Basically 
```
if(getLoweringConfig(op))` {
  ...
}
```

is same as `isLoweringConfigSet`. There is too much of chained conditioning here, which is really hard to read and really hard to modify. In many ways the configuration attributes are really to avoid this kind of code-pattern. The steps are

1) Check if configuration is already set. If so do nothing.
2) Set configuration in the IR. Whether you have set the configuration or not is specified in the IR instead of a side-car data-structure.

This is basically the thumb rule of materialize state in IR, instead of c++ code. You dont need the chain of `isTensorCoreSupported` -> `getTensorCoreConfig` -> `setTensorCoreConfig`. If tensor core is supported, you set the config in the IR directly. There is no need to add `TileWorkgroupSizePair` data structure here. You just call the `setOpConfigAndEntryPointFnTranslation` and this will put all the information in the right place.